SPECS/glusterfs/glusterfs.spec

```diff
@@ -34,463 +45,666 @@
 %global _without_server --without-server
+%endif
+# syslog
+# if you wish to build rpms without syslog logging, compile like this
+# rpmbuild -ta @PACKAGE_NAME@-@PACKAGE_VERSION@.tar.gz --without syslog
+%{?_without_syslog:%global _without_syslog --disable-syslog}
 # disable syslog forcefully as rhel <= 6 doesn't have rsyslog or rsyslog-mmcount
 # Fedora deprecated syslog, see
 #  https://fedoraproject.org/wiki/Changes/NoDefaultSyslog
 # (And what about RHEL7?)
+%if ( 0%{?fedora} && 0%{?fedora} >= 20 ) || ( 0%{?rhel} && 0%{?rhel} <= 6 )
 %global _without_syslog --disable-syslog
+%endif
 # tsan
 # if you wish to compile an rpm with thread sanitizer...
 # rpmbuild -ta @PACKAGE_NAME@-@PACKAGE_VERSION@.tar.gz --with tsan
 %{?_with_tsan:%global _with_tsan --enable-tsan}
+%if ( 0%{?rhel} && 0%{?rhel} < 7 )
+%global _with_tsan %{nil}
+%endif
 # valgrind
 # if you wish to compile an rpm to run all processes under valgrind...
 # rpmbuild -ta @PACKAGE_NAME@-@PACKAGE_VERSION@.tar.gz --with valgrind
 %{?_with_valgrind:%global _with_valgrind --enable-valgrind}
 ##-----------------------------------------------------------------------------
 ## All %%global definitions should be placed here and keep them sorted
 ##
+# selinux booleans whose defalut value needs modification
+# these booleans will be consumed by "%%selinux_set_booleans" macro.
+%if ( 0%{?rhel} && 0%{?rhel} >= 8 )
+%global selinuxbooleans rsync_full_access=1 rsync_client=1
+%endif
+%if ( 0%{?fedora} ) || ( 0%{?rhel} && 0%{?rhel} > 6 ) || ( 0%{?mariner} && 0%{?mariner} >= 3 )
+%global _with_systemd true
+%endif
+%if ( 0%{?fedora} ) || ( 0%{?rhel} && 0%{?rhel} >= 7 ) || ( 0%{?mariner} && 0%{?mariner} >= 3 )
 %global _with_firewalld --enable-firewalld
-%if ( 0%{?_tmpfilesdir:1} )
+%endif
+%if 0%{?_tmpfilesdir:1}
 %global _with_tmpfilesdir --with-tmpfilesdir=%{_tmpfilesdir}
 %else
 %global _with_tmpfilesdir --without-tmpfilesdir
 %endif
 # without server should also disable some server-only components
-%if ( 0%{?_without_server:1} )
+%if 0%{?_without_server:1}
 %global _without_events --disable-events
 %global _without_georeplication --disable-georeplication
+%global _without_linux_io_uring --disable-linux-io_uring
 %global _with_gnfs %{nil}
 %global _without_ocf --without-ocf
 %endif
+%if ( 0%{?fedora} ) || ( 0%{?rhel} && 0%{?rhel} > 7 ) || ( 0%{?mariner} && 0%{?mariner} >= 3 )
+%global _usepython3 1
 %global _pythonver 3
-%global service_start()   /bin/systemctl --quiet start %{1}.service || : \
+%else
+%global _usepython3 0
+%global _pythonver 2
+%endif
+# From https://fedoraproject.org/wiki/Packaging:Python#Macros
+%if ( 0%{?rhel} && 0%{?rhel} <= 6 ) && ( 0%{!?mariner} )
+%{!?python2_sitelib: %global python2_sitelib %(python2 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())")}
+%{!?python2_sitearch: %global python2_sitearch %(python2 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib(1))")}
```

Comment:
Remove **ALL** references to Python2. We've had a major clean-up for that in 2.0 and don't want to bring that back in.

Reply:
i don't understand why we are making changes to upstream bits.  it will just make life harder for the next upgrader.

Reply:
removed (debate can continue, but want to get this merged)